enginetest/join_planning_tests.go

```diff
@@ -651,56 +657,21 @@ func evalJoinOrder(t *testing.T, harness Harness, e *sqle.Engine, q string, exp
 }
 
 func collectJoinOrder(n sql.Node) []string {
-	var order []string
-
-	j, ok := n.(*plan.JoinNode)
-	if ok {
-		if n, ok := j.Left().(sql.NameableNode); ok {
-			order = append(order, n.Name())
-		}
-	}
-	for _, n := range n.Children() {
-		order = append(order, collectJoinOrder(n)...)
-	}
-	if ok {
-		switch n := j.Right().(type) {
-		case sql.NameableNode:
-			order = append(order, n.Name())
-		case *plan.HashLookup:
-			ok = false
-			r := n.Child
-			for !ok {
-				switch n := r.(type) {
-				case sql.NameableNode:
-					order = append(order, n.Name())
-					ok = true
-				case *plan.JoinNode:
-					ok = true
-				case *plan.Distinct:
-					r = n.Child
-				case *plan.Project:
-					r = n.Child
-				case *plan.CachedResults:
-					r = n.Child
-				default:
-					ok = true
-				}
-			}
+	order := []string{}
+
+	switch n := n.(type) {
+	case *plan.JoinNode:
+		order = append(order, collectJoinOrder(n.Left())...)
+		order = append(order, collectJoinOrder(n.Right())...)
+	case *plan.TableAlias:
```

Comment:
Old version was pretty bad, but in the general case this misses three additional cases for `ResolvedTable`, `IndexedTableAccess`, and `SubqueryAlias`. The deleted expressioner block will also miss Subquery expressions. I guess none of our current tests hit those cases anymore.

Reply:
Yeah, this doesn't handle quite as many cases, but as you noted, we don't seem to be exercising any of those cases in these test.
I'm inclined to keep the simpler implementation for now and expanding on it when we need to.